mgr/nfs: include placement details and active/passive roles in cluster info

Shubhaj1810 opened the pull request:
**### Summary**

This PR improves the ceph nfs cluster info command output to provide clearer visibility into NFS cluster layout, deployment type, placement configuration, and daemon status.

It also updates the default ingress mode for NFS clusters to haproxy-protocol to preserve client IP addresses for correct export enforcement.

**### Problem Statement**

Previously, ceph nfs cluster info:

- Did not include placement details
- Did not show daemon runtime status
- Did not indicate whether the cluster was deployed as:
1. Active–Active
2. Active–Passive
3. Standalone

- Required users to inspect ceph orch ls to infer deployment type
- Defaulted ingress mode to haproxy-standard, which does not preserve client IP addresses

This made it difficult to understand cluster layout and enforce IP-based export restrictions.

Fixes: https://tracker.ceph.com/issues/74239
Fixes: https://tracker.ceph.com/issues/74240
Fixes: https://tracker.ceph.com/issues/74260

Signed-off-by: Shubha Jain [SHUBHA.JAIN1@ibm.com]

**### Changes Introduced**
**1. Enhanced ceph nfs cluster info Output**

1. The command now includes:

deployment_type:

- standalone
- active-active
- active-passive

2. placement (from NFS service spec)
3. status for each backend daemon
4. Explicit ingress_mode
5. Deterministic backend ordering

Example:

`{
  "mycluster": {
    "deployment_type": "active-active",
    "virtual_ip": "10.0.1.100",
    "backend": [
      {
        "hostname": "node1",
        "ip": "10.0.1.10",
        "port": 2049,
        "status": "running"
      }
    ],
    "placement": {
      "count": 3
    },
    "ingress_mode": "haproxy-protocol",
    "ingress_port": 2049,
    "port": 2049,
    "ingress_monitor_port": 9000,
    "monitor_port": 9000
  }
}
`



## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ x ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ x ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ x ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ x ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)

You must only issue one Jenkins command per-comment. Jenkins does not understand
comments with more than one command.
</details>

ShwetaBhosale1 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,169 @@ def list_nfs_cluster(self) -> List[str]:
             log.exception("Failed to list NFS Cluster")
             raise ErrorResponse.wrap(e)
 
+    def _get_daemon_role(
```
We don't need to show daemon roles(active and standby) as standby doesn't make much sense here. We can still show daemon status here. 
And we need to show 'cluster mode' as active-active or active-passive.

Shubhaj1810 commented:
Testing Logs:
After Fix:

{
  "nfs-cephfs": {
    "deployment_type": "active-passive",
    "virtual_ip": "10.8.130.191",
    "backend": [
      {
        "hostname": "ceph-node-00",
        "ip": "192.168.100.100",
        "port": 2049,
        "status": "running"
      },
      {
        "hostname": "ceph-node-01",
        "ip": "192.168.100.101",
        "port": 2049,
        "status": "running"
      },
      {
        "hostname": "ceph-node-02",
        "ip": "192.168.100.102",
        "port": 2049,
        "status": "running"
      }
    ],
    "placement": {
      "count": 3
    },
    "ingress_mode": "haproxy-protocol",
    "port": 2049,
    "monitor_port": 9049
  }
}

ceph-jenkins commented:
Can one of the admins verify this patch?

ShwetaBhosale1 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,154 @@ def list_nfs_cluster(self) -> List[str]:
+        - "unknown": Status cannot be determined
+        """
+        from orchestrator import DaemonDescriptionStatus
+        
```
Use to_str method of DaemonDescriptionStatus to return str.

ShwetaBhosale1 commented on src/pybind/mgr/orchestrator/module.py:
```diff
@@ -224,7 +224,9 @@ class IngressType(enum.Enum):
-            return IngressType(self.haproxy_standard)
+            # Default to haproxy-protocol to preserve client IP addresses
+            # for proper IP-level export restrictions in NFS Ganesha
+            return IngressType(self.haproxy_protocol)
```
Please add the change for updating the default ingress mode in a separate commit.

ShwetaBhosale1 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,154 @@ def list_nfs_cluster(self) -> List[str]:
+        nfs_services = orchestrator.raise_if_exception(nfs_sc)
+        for svc in nfs_services:
+            if svc.spec.service_id == cluster_id:
+                placement = svc.spec.placement
```
To show placement you can just call to_json method of PlacementSpec

Shubhaj1810 replied:
Acknowledged. Placement information is now returned via PlacementSpec.to_json(), which keeps the output consistent and avoids manual field handling.

Shubhaj1810 commented on src/pybind/mgr/dashboard/ci/cephadm/ceph_cluster.yml:
```diff
@@ -5,7 +5,7 @@ parameters:
  numcpus: 1
  memory: 2048
- image: fedora42
+ image: fedora40
```
No, i don't have the fedora42 image installed locally on my system and instead i'm using fedora40

Shubhaj1810 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,154 @@ def list_nfs_cluster(self) -> List[str]:
+            # With HAProxy ingress, NFS daemons are load-balanced (active-active behind the scenes)
+            # but we present it as active-passive for operational clarity since clients
+            # connect through a single virtual IP
+            deployment_type = "active-passive"
```
Acknowledged. Deployment type is now determined based on placement/daemon count and correctly reports active-active when multiple NFS daemons are present.

Shubhaj1810 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,154 @@ def list_nfs_cluster(self) -> List[str]:
+        if ingress_port:
+            r['port'] = ingress_port
+        if monitor_port:
+            r['monitor_port'] = monitor_port
```
Fixed — the result keys now use ingress_port and ingress_monitor_port.

ShwetaBhosale1 replied:
Ah, I hadn’t noticed the earlier behavior. The ingress ports will be available only for HAProxy, so you can keep the old behavior here, as it doesn’t make sense to have two fields for the same port.

Shubhaj1810 committed:
mgr/orchestrator: default NFS ingress to haproxy-protocol mode

Change default ingress mode from haproxy-standard to haproxy-protocol
to preserve client IP addresses for proper IP-level export restrictions
in NFS Ganesha.

Fixes: https://tracker.ceph.com/issues/74260

Signed-off-by: Shubha Jain <SHUBHA.JAIN1@ibm.com>

ShwetaBhosale1 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,139 @@ def list_nfs_cluster(self) -> List[str]:
+                        monitor_port = svc.ports[1]
+                break
+
+        # No longer needed as we're not showing active/passive per daemon
```
remove this comment as its not needed

ShwetaBhosale1 commented on src/pybind/mgr/orchestrator/module.py:
```diff
@@ -2603,3 +2605,4 @@ def _update_service(self, service_type: NonCephImageServiceTypes, image: str) ->
         completion = self.update_service(service_type.value, service_type.name, image)
         raise_if_exception(completion)
         return HandleCommandResult(stdout=completion.result_str())
+
```
Remove this extra line

Shubhaj1810 commented:
> Also, please include the output of this command both with and without ingress configured.

1. Standalone NFS (No Ingress)
```
ceph nfs cluster create mynfs "1 host1"
$ ceph nfs cluster info mynfs
{
  "mynfs": {
    "deployment_type": "standalone",
    "virtual_ip": null,
    "backend": [
      {
        "hostname": "host1",
        "ip": "10.0.0.10",
        "port": 2049,
        "status": "running"
      }
    ],
    "placement": {
      "count": 1
    }
  }
}
```

2. Active-Active with HAProxy Protocol (Multiple Daemons)
```
ceph nfs cluster create mynfs-ha "3 label:nfs" \
  --ingress \
  --virtual-ip 10.0.0.100/24 \
  --ingress-mode haproxy-protocol
  
$ ceph nfs cluster info mynfs-ha
{
  "mynfs-ha": {
    "deployment_type": "active-active",
    "virtual_ip": "10.0.0.100",
    "backend": [
      {
        "hostname": "host1",
        "ip": "10.0.0.10",
        "port": 12049,
        "status": "running"
      },
      {
        "hostname": "host2",
        "ip": "10.0.0.11",
        "port": 12049,
        "status": "running"
      },
      {
        "hostname": "host3",
        "ip": "10.0.0.12",
        "port": 12049,
        "status": "running"
      }
    ],
    "placement": {
      "count": 3,
      "label": "nfs"
    },
    "ingress_mode": "haproxy-protocol",
    "ingress_port": 2049,
    "ingress_monitor_port": 9049
  }
}
```

3. Active-Passive with HAProxy Standard (Single Daemon)
```
ceph nfs cluster create mynfs-passive "1 host1" \
  --ingress \
  --virtual-ip 10.0.0.100/24 \
  --ingress-mode haproxy-standard

$ ceph nfs cluster info mynfs-passive

{
  "mynfs-passive": {
    "deployment_type": "active-passive",
    "virtual_ip": "10.0.0.100",
    "backend": [
      {
        "hostname": "host1",
        "ip": "10.0.0.10",
        "port": 12049,
        "status": "running"
      }
    ],
    "placement": {
      "count": 1
    },
    "ingress_mode": "haproxy-standard",
    "ingress_port": 2049,
    "ingress_monitor_port": 9049
  }
}
```

4. Keepalive-Only Mode (Direct Connection)
```
ceph nfs cluster create mynfs-keepalive "1 host1" \
  --ingress \
  --virtual-ip 10.0.0.100/24 \
  --ingress-mode keepalive-only
  
$ ceph nfs cluster info mynfs-keepalive

{
  "mynfs-keepalive": {
    "deployment_type": "active-passive",
    "virtual_ip": "10.0.0.100",
    "backend": [
      {
        "hostname": "host1",
        "ip": "10.0.0.10",
        "port": 2049,
        "status": "running"
      }
    ],
    "placement": {
      "count": 1
    },
    "ingress_mode": "keepalive-only",
    "ingress_port": 2049
  }
}
```

5. Default Ingress (Now HAProxy Protocol)
```
ceph nfs cluster create mynfs-default "2 host1,host2" \
  --ingress \
  --virtual-ip 10.0.0.100/24
# Note: No --ingress-mode specified, uses default (now haproxy-protocol)

$ ceph nfs cluster info mynfs-default

{
  "mynfs-default": {
    "deployment_type": "active-active",
    "virtual_ip": "10.0.0.100",
    "backend": [
      {
        "hostname": "host1",
        "ip": "10.0.0.10",
        "port": 12049,
        "status": "running"
      },
      {
        "hostname": "host2",
        "ip": "10.0.0.11",
        "port": 12049,
        "status": "running"
      }
    ],
    "placement": {
      "count": 2,
      "hosts": ["host1", "host2"]
    },
    "ingress_mode": "haproxy-protocol",
    "ingress_port": 2049,
    "ingress_monitor_port": 9049
  }
}
```

ShwetaBhosale1 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,137 @@ def list_nfs_cluster(self) -> List[str]:
+        - "starting": Daemon is starting up
+        - "unknown": Status cannot be determined
+        """
+        from orchestrator import DaemonDescriptionStatus
```
orchestrator is already imported in this file (line 11), please use the existing import

ShwetaBhosale1 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,137 @@ def list_nfs_cluster(self) -> List[str]:
+        Retrieve and format NFS cluster information including daemon status,
+        placement, and ingress configuration.
+        """
+        from orchestrator import DaemonDescriptionStatus
```
orchestrator is already imported in this file (line 11), please use the existing import

Shubhaj1810 replied:
> orchestrator is already imported, please use the existing import

Fixed. Removed the local import and now using orchestrator.DaemonDescriptionStatus.to_str() directly via the existing import

Shubhaj1810 commented on src/pybind/mgr/nfs/tests/test_nfs.py:
```diff
@@ -1231,7 +1231,12 @@ def _do_test_cluster_info(self):
+            "virtual_ip": None,
+            "backend": [],
+            "placement": None
+        }}
```
The code now returns {} for placement instead of None when unset. I’ve updated the test to match this behavior.

ShwetaBhosale1 commented:
It looks like the latest commit isn’t signed. Also, could you please squash the commits and keep just two: one for the cluster info and one for changing the default ingress mode?

adk3798 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -227,57 +227,133 @@ def list_nfs_cluster(self) -> List[str]:
+        - "starting": Daemon is starting up
+        - "unknown": Status cannot be determined
+        """
+        return orchestrator.DaemonDescriptionStatus.to_str(daemon_status)
```
given DaemonDescriptionStatus is defined in the orchestrator code (meaning we could change it there and forget to update this comment) and this function is only one line, I think we can just directly call `orchestrator.DaemonDescriptionStatus.to_str(daemon_status)` rather than having this helper function here

Shubhaj1810 committed:
mgr/nfs: improve cluster info implementation and fix lint regressions

- Improve cluster info implementation
- Fix deployment type logic
- Resolve lint issues for check jobs

Signed-off-by: Shubha jain <SHUBHA.JAIN1@ibm.com>

adk3798 commented on src/pybind/mgr/nfs/cluster.py:
```diff
@@ -228,56 +228,132 @@ def list_nfs_cluster(self) -> List[str]:
+            'virtual_ip': virtual_ip,
+            'backend': backends,
+            'placement': placement.to_json() if placement else {},
+        }
```
There's a teuthology test failures in my run related to this change to the format
```
2026-02-14T00:23:43.704 INFO:tasks.cephfs_test_runner:
2026-02-14T00:23:43.704 INFO:tasks.cephfs_test_runner:======================================================================
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:FAIL: test_cluster_info (tasks.cephfs.test_nfs.TestNFS.test_cluster_info)
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:Test cluster info outputs correct ip and hostname
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_adk3798_ceph_632a9804aab65871427145f015001eb516b59504/qa/tasks/cephfs/test_nfs.py", line 830, in test_cluster_info
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:    self.assertDictEqual(info_output, host_details)
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:AssertionError: {'tes[49 chars] 2049, 'status': 'running'}], 'deployment_type[59 chars]one}} != {'tes[49 chars] 2049}], 'virtual_ip': None}}
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:- {'test': {'backend': [{'hostname': 'trial179',
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:+ {'test': {'backend': [{'hostname': 'trial179', 'port': 2049}],
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:?                                               ++++++++++++++++
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:
2026-02-14T00:23:43.705 INFO:tasks.cephfs_test_runner:-                        'port': 2049,
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:-                        'status': 'running'}],
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:-           'deployment_type': 'standalone',
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:-           'placement': {'count': 1},
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:            'virtual_ip': None}}
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:Ran 4 tests in 449.404s
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:FAILED (failures=1)
2026-02-14T00:23:43.706 INFO:tasks.cephfs_test_runner:
2026-02-14T00:23:43.708 INFO:tasks.cephfs_test_runner:======================================================================
2026-02-14T00:23:43.708 INFO:tasks.cephfs_test_runner:FAIL: test_cluster_info (tasks.cephfs.test_nfs.TestNFS.test_cluster_info)
2026-02-14T00:23:43.708 INFO:tasks.cephfs_test_runner:Test cluster info outputs correct ip and hostname
2026-02-14T00:23:43.708 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2026-02-14T00:23:43.708 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_adk3798_ceph_632a9804aab65871427145f015001eb516b59504/qa/tasks/cephfs/test_nfs.py", line 830, in test_cluster_info
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:    self.assertDictEqual(info_output, host_details)
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:AssertionError: {'tes[49 chars] 2049, 'status': 'running'}], 'deployment_type[59 chars]one}} != {'tes[49 chars] 2049}], 'virtual_ip': None}}
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:- {'test': {'backend': [{'hostname': 'trial179',
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:+ {'test': {'backend': [{'hostname': 'trial179', 'port': 2049}],
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:?                                               ++++++++++++++++
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:-                        'port': 2049,
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:-                        'status': 'running'}],
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:-           'deployment_type': 'standalone',
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:-           'placement': {'count': 1},
2026-02-14T00:23:43.709 INFO:tasks.cephfs_test_runner:            'virtual_ip': None}}
2026-02-14T00:23:43.710 INFO:tasks.cephfs_test_runner:
2026-02-14T00:23:43.710 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_teuthology_8bec0da71becad44414c54979f64c9ef0e7099c6/teuthology/run_tasks.py", line 112, in run_tasks
    manager.__enter__()
  File "/usr/lib/python3.12/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/home/teuthworker/src/github.com_adk3798_ceph_632a9804aab65871427145f015001eb516b59504/qa/tasks/cephfs_test_runner.py", line 211, in task
    raise RuntimeError("Test failure: {0}".format(", ".join(bad_tests)))
RuntimeError: Test failure: test_cluster_info (tasks.cephfs.test_nfs.TestNFS.test_cluster_info)
```
you'd need to change https://github.com/ceph/ceph/blob/main/qa/tasks/cephfs/test_nfs.py#L809 to look for this new format

rkachach commented:
jenkins retest this please

Shubhaj1810 commented:
jenkins test make check

Shubhaj1810 commented:
jenkins test make check

rkachach commented:
Teuthology run was successful. Failing Test cases are not related to this PR:
https://pulpito.ceph.com/rkachach-2026-06-29_19:11:54-orch:cephadm-wip-rkachach-testing-2026-06-29-1345-distro-default-trial/
